internal/security/setup/option/optionHelpers.go

```diff
@@ -214,20 +233,64 @@ func markComplete(dirPath string) error {
 	return nil
 }
 
-func getXdgRuntimeDir() string {
+func getWorkDir() string {
+	var workDir string
+	var err error
+
 	if xdgRuntimeDir, ok := os.LookupEnv(envXdgRuntimeDir); ok {
-		return xdgRuntimeDir
+		workDir = filepath.Join(xdgRuntimeDir, pkiInitBaseDir)
+	} else if setup.Configuration.SecretsSetup.WorkDir != "" {
```

Comment:
eventually this should also check consul/registry for the settings, but that's not for this PR

Reply:
agree, let's put that into future enhancement for using registry

Reply:
This utility will likely run before the registry is up.  It will definitely run before the registry is up if we generate a TLS certificate for the registry using this tool.

Reply:
My understanding was that we will not be generating a TLS certificate for the registry in the short or medium term, so I'd rather see us be consistent with storing configuration in files and also optionally in consul, then deal with the problem of bootstrapping consul later. 

Is there another reason why this utility would run before consul other than generating a TLS certificate for the registry some day?

Reply:
Service-to-service encryption.  If Consul Connect were used to provide that using a Vault back-end as a certificate provider, that would also affect the start order by having Consul depend on Vault rather than Vault depend on Consul, or having two Consul instances.  Again, no design implied or proposed -- just possibility thinking here.

Reply:
Hmm, well regardless let's file an issue about this so we don't lose this discussion and can bring the issue to a wider audience than just this PR.

Reply:
issue filed here https://github.com/edgexfoundry/edgex-go/issues/1889